innkeeper/client.go

```diff
@@ -320,6 +318,43 @@ func getHttpError(r *http.Response) (error, bool) {
 	}
 }
 
+func (c *Client) writeRoute(url string, route *routeData) error {
+
+	res, err := json.Marshal(route)
+
+	req, err := http.NewRequest("POST", url, bytes.NewReader(res))
+
+	if err != nil {
+		return err
+	}
+
+	authToken, err := c.opts.Authentication.GetToken()
```

Comment:
As the client accepts anything that implements the Authentication interface, this is the contract between the two components. Of course, the token can be valid or not and we take this into account by returning a proper error.